contracts/NameWrapper.sol

```diff
@@ -280,6 +281,49 @@ contract NameWrapper is Ownable, ERC1155Fuse, INameWrapper {
         emit NameWrapped(ETH_NODE, label, wrappedOwner, _fuses);
     }
 
+    /**
+     * @dev Registers a new .eth second-level domain and wraps it.
+     *      Only callable by authorised controllers.
+     * @param label The label to register (Eg, 'foo' for 'foo.eth').
+     * @param wrappedOwner The owner of the wrapped name.
+     * @param duration The duration, in seconds, to register the name for.
+     * @param resolver The resolver address to set on the ENS registry (optional).
+     * @return expires The expiry date of the new name, in seconds since the Unix epoch.
+     */
+    function registerAndWrapETH2LD(
```

Comment:
Would this function also allow us to set up records like the address record like the current controller would? Would that be setup in the controller that "calls" this NameWrapper?

Reply:
Right - I think we discussed and realised that having this function call resolver stuff wouldn't be any more efficient than the caller doing that.